cpp/include/graph.hpp

```diff
@@ -661,4 +663,179 @@ template <typename ET>
 struct invalid_edge_id : invalid_idx<ET> {
 };
 
+namespace mg {
+/**
+ * @brief              A Replicatable Graph representation to replicate
+ *                     a graph using the handle
+ *
+ * @tparam VT              Type of vertex id
+ * @tparam ET              Type of edge id
+ * @tparam WT              Type of weight
+ * @tparam GraphTypeView   Type of Graph View
+ */
+template <typename VT, typename ET, typename WT, typename GraphTypeView>
+class ReplicatableGraph {
+ public:
+  /**
+   * @brief                           Allows broadcast of a Graph via communicator and handle
+   * required memory
+   *
+   * @param[in] handle                Library handle (RAFT). If a communicator is expected
+   * to be initialized already
+   * @param[in]  graph_to_replicate  Pointer to the graph, should be null if rank != 0
+   * != 0
+   */
+  ReplicatableGraph(const raft::handle_t &handle, GraphTypeView const *graph_to_replicate)
+  {
+    handle_   = &handle;
+    rank_     = handle_->get_comms().get_rank();
```

Comment:
Perhaps there should be a check to see if comms are set before calling get_comms()

Reply:
Added a check in eab569b